src/inausoft.netCLI/netCLI.cs

```diff
@@ -0,0 +1,99 @@
+            {
+                throw new ArgumentNullException($"{nameof(services)} cannot be null.");
+            }
+
+            CLIConfiguration configuration = new CLIConfiguration();
+
+            setup(configuration);
+
+            services.AddSingleton(configuration);
+
+            configuration._commandMap.Values.ToList().ForEach(it =>
+            {
+                services.AddSingleton(it);
+            });
+
+            return services;
+        }
+
+        /// <summary>
+        /// Runs <see cref="ICommandHandler"/> for command specified in args./>
+        /// </summary>
+        /// <param name="serviceProvider"></param>
+        /// <param name="args">Application's command line arguments.</param>
+        /// <returns></returns>
+        public static int RunCLI(this IServiceProvider serviceProvider, string[] args)
+        {
+            var config = serviceProvider.GetService<CLIConfiguration>();
+
+            if (config == null)
+            {
+                throw new InvalidOperationException($"{nameof(CLIConfiguration)} was not registered. Run '{nameof(AddCLI)}' first.");
+            }
+
+            var commandType = config._commandMap.Keys.FirstOrDefault(it =>
+                Attribute.IsDefined(it, typeof(CommandAttribute)) &&
+                (Attribute.GetCustomAttribute(it, typeof(CommandAttribute)) as CommandAttribute).Name == args[0]);
+
+            if (commandType == null)
+            {
+                throw new InvalidCommandException(args[0], $"Command {args[0]} was not mapped.");
+            }
+
+            var command = config.Deserializer.Deserialize(commandType, string.Join(" ", args.Skip(1)));
+
+            var handler = serviceProvider.GetRequiredService(config._commandMap[command.GetType()]) as ICommandHandler;
+
+            return handler.Run(command);
+        }
+
+        public static int RunCLI(CLIConfiguration config, string[] args, params ICommandHandler[] handlers)
+        {
+            if (config == null)
+            {
+                throw new InvalidOperationException($"{nameof(CLIConfiguration)} was not registered. Run '{nameof(AddCLI)}' first.");
+            }
+
+            var commandType = config._commandMap.Keys.FirstOrDefault(it =>
+                Attribute.IsDefined(it, typeof(CommandAttribute)) &&
+                (Attribute.GetCustomAttribute(it, typeof(CommandAttribute)) as CommandAttribute).Name == args[0]);
+
+            if (commandType == null)
+            {
+                throw new InvalidCommandException(args[0], $"Command {args[0]} was not mapped.");
+            }
+
+            var command = config.Deserializer.Deserialize(commandType, string.Join(" ", args.Skip(1)));
+
+            var handler = handlers.First(it => it.GetType() == config._commandMap[command.GetType()]) as ICommandHandler;
+
+            return handler.Run(command);
+        }
+
+        public static CLIConfiguration MapHelpCommand(this CLIConfiguration config)
```

Comment:
I would consider add also VersionCommand

Reply:
Agree. This is a subject of another PR.